src/hb-ot-var-common.hh

```diff
@@ -437,6 +437,17 @@ struct byte_data_t
   { return p == o.p && length == o.length; }
 
   explicit operator bool () const { return length; }
+
+  byte_data_t* copy (hb_serialize_context_t *c) const
+  {
+    if (unlikely (!length || !p)) return nullptr;
+
+    char* ret = c->allocate_size<char> (length);
+    if (unlikely (!ret)) return nullptr;
+    hb_memcpy (ret, p, length);
+    // cast returning pointer, required by serializer copy()
+    return reinterpret_cast<byte_data_t *> (ret);
```

Comment:
But this line is wrong! Better to return nullptr. No?

Reply:
Can you return void here and call the copy() directly, instead of calling it through the serializer?